### PR TITLE
Add validatessl option to enable/disable self-signed certificates

### PR DESCRIPTION
**Pull request by rcolinray**

### CODE REVIEW

File: src/ios/CDVInAppBrowser.h [hunk 1]
```diff
@@ -63,24 +63,31 @@
...
 @property (nonatomic, assign) BOOL hidden;
 @property (nonatomic, assign) BOOL disallowoverscroll;
+@property (nonatomic, assign) BOOL validatessl;
 
 + (CDVInAppBrowserOptions*)parseOptions:(NSString*)options;
 
 @end
 
-@interface CDVInAppBrowserViewController : UIViewController <UIWebViewDelegate, CDVScreenOrientationDelegate>{
+@interface CDVInAppBrowserViewController : UIViewController <UIWebViewDelegate, CDVScreenOrientationDelegate, NSURLConnectionDataDelegate>{
```
Comment (war1025) [inline src/ios/CDVInAppBrowser.h:72]: Should we line-wrap this?

**Comment by war1025** [comment]
+1. Objective C looks funny to me.
 - [x] Maybe break things up so they fit on github better

File: src/ios/CDVInAppBrowser.h (hunk 1 above)
Comment (rcolinray) [reply src/ios/CDVInAppBrowser.h:72]: The approach in the iOS community seems to be to let lines run as long as needed and let Xcode do the formatting. I know with some of our other forks that we tend to keep to the style of the original code. Not really hard over either way though, and I could definitely see line-wrapping it to make it look nicer on Github.

File: src/ios/CDVInAppBrowser.m [hunk 2]
```diff
@@ -887,10 +891,21 @@ - (void)webViewDidStartLoad:(UIWebView*)theWebView
...
+    BOOL isSecuredUrl = [[request.URL scheme] isEqualToString:@"https"];
 
     if (isTopLevelNavigation) {
         self.currentURL = request.URL;
     }
+
+    if (isSecuredUrl && self.validateSsl == NO) {
+        // iOS disallows untrusted certs, so we must use NSURLConnection to do custom authentication.
+        // We cancel the original request and store it until authentication is completed, then
+        // reload it.
```
Comment (aronbierbaum) [inline src/ios/CDVInAppBrowser.m]: I don't think you mean `authentication` here.

File: src/ios/CDVInAppBrowser.m [hunk 3]
```diff
@@ -937,6 +952,25 @@ - (void)webView:(UIWebView*)theWebView didFailLoadWithError:(NSError*)error
     [self.navigationDelegate webView:theWebView didFailLoadWithError:error];
 }
 
+#pragma mark NSURLConnectionDataDelegate
+
+- (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge {
```
Comment (aronbierbaum) [inline src/ios/CDVInAppBrowser.m]: Could you provide a high level description of what is happening here? Are we jacking into an authentication request to store the base URL for usage later?

File: src/ios/CDVInAppBrowser.m [hunk 4]
```diff
@@ -887,10 +891,21 @@ - (void)webViewDidStartLoad:(UIWebView*)theWebView
...
 {
     BOOL isTopLevelNavigation = [request.URL isEqual:[request mainDocumentURL]];
+    BOOL isSecuredUrl = [[request.URL scheme] isEqualToString:@"https"];
 
     if (isTopLevelNavigation) {
         self.currentURL = request.URL;
     }
+
+    if (isSecuredUrl && self.validateSsl == NO) {
+        // iOS disallows untrusted certs, so we must use NSURLConnection to do custom authentication.
```
Comment (aronbierbaum) [inline src/ios/CDVInAppBrowser.m]: We aren't doing custom authentication, so maybe this comment needs cleaned up?

File: src/ios/CDVInAppBrowser.m [hunk 5]
```diff
@@ -937,6 +952,25 @@ - (void)webView:(UIWebView*)theWebView didFailLoadWithError:(NSError*)error
     [self.navigationDelegate webView:theWebView didFailLoadWithError:error];
 }
 
+#pragma mark NSURLConnectionDataDelegate
+
+- (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticationChallenge:(NSURLAuthenticationChallenge *)challenge {
+    // Allow self-signed SSL certificates from URLs that have the same host as the base URL.
```
Comment (aronbierbaum) [inline src/ios/CDVInAppBrowser.m:958]: Is this limited to self signed certificates, or are we allowing any invalid certificates when we disable checking?

Comment (rcolinray) [reply src/ios/CDVInAppBrowser.m:958]: Any invalid certificates

**Comment by aronbierbaum** [comment]
+1

**Pull request merged**